src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -115,6 +123,8 @@ LRUFileCachePriority::remove(LRUQueue::iterator it, const CachePriorityGuard::Lo
         log, "Removed entry from LRU queue, key: {}, offset: {}, size: {}",
         entry.key, entry.offset, entry.size.load());
 
+    if (eviction_pos == it)
+        eviction_pos = std::next(it);
     return queue.erase(it);
```

Comment:
maybe
```suggestion
    auto next_it = queue.erase(it);
    if (eviction_pos == it)
    	eviction_pos = next_it;
    return next_it;
```

Reply:
Is it valid to compare `if (eviction_pos == it)` after `it` is removed from `queue`?

Reply:
wow I'm not sure
Let's leave it then